### ISSUE TITLE
V-measure returns 1 for independent nontrivial clusterings

### ISSUE THREAD

**Issue by Robertboy18** [Lightning-AI/torchmetrics#3484]
For two independent, nontrivial clusterings, both homogeneity and completeness are zero, but `v_measure_score` currently returns `1.0`.

```python
import torch
from torchmetrics.functional.clustering import v_measure_score

preds = torch.tensor([0, 1, 0, 1])
target = torch.tensor([0, 0, 1, 1])

print(v_measure_score(preds, target))  # tensor(1.)
```

scikit-learn returns `0.0` for the same labels. That also matches the weighted harmonic mean definition: when both component scores are zero, V-measure should be zero.

This reproduces on current `master` (`8d008de1`) with Python 3.12.13 and PyTorch 2.13.0. I have a small fix and regression test ready. I also checked the edge case with a small Lean formalization and can attach it if useful :)
